tests/token_tests.ts

```diff
@@ -0,0 +1,216 @@
+    })
+
+    test(`${permission} key: Search in tenant token with wildcard`, async () => {
+      const client = await getClient(permission)
+      const token = client.generateTenantToken(['*'])
+
+      const searchClient = new MeiliSearch({ host: HOST, apiKey: token })
+
+      // search
+      const { hits } = await searchClient.index(UID).search('pride')
+      expect(hits[0].title).toEqual('Pride and Prejudice')
+    })
+
+    test(`${permission} key: Search in tenant token with custom api key`, async () => {
+      // add filterable
+      const masterClient = await getClient('master')
+      const { key } = await masterClient.createKey({
+        expiresAt: null,
+        description: 'Custom key',
+        actions: ['search'],
+        indexes: [UID],
+      })
+
+      const client = await getClient(permission)
+      const token = client.generateTenantToken(['*'], { apiKey: key })
+
+      const searchClient = new MeiliSearch({ host: HOST, apiKey: token })
+
+      // search
+      const { hits } = await searchClient.index(UID).search('pride')
+      expect(hits[0].title).toEqual('Pride and Prejudice')
+    })
+
+    test(`${permission} key: Search in tenant token with expire date`, async () => {
+      const client = await getClient(permission)
+      const date = new Date('December 17, 4000 03:24:00')
+      const token = client.generateTenantToken(['*'], {
+        expiresAt: date,
+      })
+
+      const [_, payload] = token.split('.')
+      expect(JSON.parse(decode64(payload)).exp).toEqual(64090923840000)
+
+      const searchClient = new MeiliSearch({ host: HOST, apiKey: token })
+
+      // search
+      const { hits } = await searchClient.index(UID).search('pride')
+      expect(hits[0].title).toEqual('Pride and Prejudice')
+    })
+
+    test(`${permission} key: Search in tenant token with specific index set to null`, async () => {
+      const client = await getClient(permission)
+      const token = client.generateTenantToken({
+        [UID]: null,
+      })
+      const searchClient = new MeiliSearch({ host: HOST, apiKey: token })
+
+      // search
+      const { hits } = await searchClient.index(UID).search('Pride')
+      expect(hits[0].title).toEqual('Pride and Prejudice')
+    })
+
+    test(`${permission} key: Search in tenant token with specific index and specific rules`, async () => {
+      // add filterable
+      const masterClient = await getClient('master')
+      const { uid } = await masterClient
+        .index(UID)
+        .updateFilterableAttributes(['id'])
+      await masterClient.waitForTask(uid)
+
+      const client = await getClient(permission)
+      const token = client.generateTenantToken({
+        [UID]: { filter: 'id = 2' },
+      })
+      const searchClient = new MeiliSearch({ host: HOST, apiKey: token })
+
+      // search
+      const { hits } = await searchClient.index(UID).search()
+      expect(hits[0].title).toEqual('Le Rouge et le Noir')
```

Comment:
I have had this conversation with Amélie, regarding the idea of testing just the behavior, not the result itself.

You did that in the last tests when you just checked if the search result failed. IMHO you should that here as well:

```js
await expect(
  searchClient.index(UID).search('pride')
).resolves...
```

If you look closer, you are asserting the success of the usage of your brand new tenant token without having to care about what the server responded (if the first result is really the `Le Rouge et le Noir`)